### PR TITLE
RATIS-2123. Remove old releases from website

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

List only 3.0.1 and 3.1.0 in [Downloads](https://ratis.apache.org/downloads.html) page.

https://issues.apache.org/jira/browse/RATIS-2123

## How was this patch tested?

```
hugo serve
open http://localhost:1313
```

### CODE REVIEW

File: content/post/2.5.1.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 title: Release 2.5.1 is available
 date: 2023-05-05
 type: release
-linked: true
+linked: false
```
Comment (szetszwo) [inline content/post/2.5.1.md:5]: @adoroszlai , thanks a lot for working on this!  Let's keep 2.5.1?  Someone may be still using branch-2.

**Comment by szetszwo** [comment]
@adoroszlai , could you also update `3.1.0.md` as below?  It is to replace `maintenance` with `minor` and break the long line.
```patch
+++ b/content/post/3.1.0.md
@@ -18,7 +18,9 @@ linked: true
 
 [Download](https://ratis.apache.org/downloads.html)
 
-This is a maintenance release with multiple improvements and bugfixes. See the [changes between 3.0.1 and 3.1.0](https://github.com/apache/ratis/compare/ratis-3.0.1...ratis-3.1.0) releases.
+This is a minor release with multiple improvements and bugfixes.
+See the [changes between 3.0.1 and 3.1.0](https://github.com/apache/ratis/compare/ratis-3.0.1...ratis-3.1.0) releases.
```

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by adoroszlai** [comment]
Thanks @szetszwo for the review.

**Pull request merged**